p2p/net/udp.go

```diff
@@ -81,14 +90,76 @@ func newUDPListener(listenAddress *net.UDPAddr) (*net.UDPConn, error) {
 
 var IPv4LoopbackAddress = net.IP{127, 0, 0, 1}
 
+func (n *UDPNet) initSession(to string, remote p2pcrypto.PublicKey) (NetworkSession, error) {
+	raddr, err := resolveUDPAddr(to)
+	if err != nil {
+		return nil, err
+	}
+
+	session := createSession(n.local.PrivateKey(), remote)
+	handshakeMessage, err := generateHandshakeMessage(session, n.config.NetworkID, 1010, n.local.PublicKey())
```

Comment:
Also, is it the same handshake message that will be sent over a tcp session?

Reply:
you're right about the configuration and port. I'll fix that. in fact, the port is not used in the message right now since we exchange it in the "Ping" message.
same handshake message for now indeed. when we'll want to optimize discovery we can tailor the handshake message to also send discovery related stuff. but the key-exchange is the same.